networkx/generators/tests/test_random_graphs.py

```diff
@@ -308,10 +308,19 @@ def root(u, w, r):
             return r / c + w
 
         c = 1
-        graph = nx.random_kernel_graph(1000, integral, root)
         graph = nx.random_kernel_graph(1000, integral, root, seed=42)
         assert len(graph) == 1000
 
+    def test_random_kernel_graph_default_root(self):
+        pytest.importorskip("scipy")
+
+        def integral(u, w, z):
+            return c * (z - w)
+
+        c = 1
```

Comment:
I realize this was just made to be consistent with the test from which it was derived, but let's go ahead and clean up the unused parameter (especially one that crosses scope boundaries)

```suggestion
        def integral(u, w, z):
            return (z - w)

```